mule-extensions-api/src/main/java/org/mule/runtime/extension/api/notification/NotificationHandler.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.notification;
+
+import static org.mule.runtime.extension.api.notification.ExtensionNotification.registerAction;
+
+/**
+ * This handler allows firing {@link org.mule.runtime.api.notification.Notification Notifications} with both custom data and
+ * default one.
+ *
+ * @since 1.1
+ */
+public interface NotificationHandler {
```

Comment:
Handler prefix is often used for consumers in the Mule code base. This is not a consumer but a producer. I would rename this to NotificationEmitter